src/main/scala/com/scylladb/migrator/writers/DynamoStreamReplication.scala

```diff
@@ -62,54 +51,43 @@ object DynamoStreamReplication {
 
           newMap.putAll(rec.getDynamodb.getKeys)
 
-          val row =
-            Row.fromSeq {
-              typeConverter
-                .readInternalRow(ItemUtils.toItem(newMap))
-                .toSeq(inferredSchema)
-                .map {
-                  case s: UTF8String => s.toString
-                  case x             => x
-                } ++
-                Seq(
-                  rec.getEventName match {
-                    case "INSERT" | "MODIFY" => ColumnSchema.PutOperation
-                    case "REMOVE"            => ColumnSchema.DeleteOperation
-                  }
-                )
+          val operationType =
+            rec.getEventName match {
+              case "INSERT" | "MODIFY" => putOperation
+              case "REMOVE"            => deleteOperation
             }
-
-          Some(row)
+          newMap.put(ColumnSchema.OperationTypeColumn, operationType)
+          // FIXME transform UTF8String to plain Strings
+          Some(newMap)
 
         case _ => None
       }
       .foreachRDD { msgs =>
-        val df = spark
-          .createDataFrame(
-            msgs
-              .collect {
-                case Some(row) => row
-              },
-            inferredSchema.add(ColumnSchema.OperationTypeColumn, BooleanType)
-          )
-          .repartition(partitionColumns: _*)
+        val rdd = msgs.collect {
+          // FIXME Check that this is the correct way to create a key
+          case Some(item) => (new Text(), new DynamoDBItemWritable(item))
+        }
```

Comment:
Is there a chance we can get rid of our fork of kinesis?
And use upstream?  ( https://docs.aws.amazon.com/streams/latest/dev/developing-consumers-with-kcl.html ? or https://docs.aws.amazon.com/streams/latest/dev/developing-consumers-with-kcl-v2.html ?)

Reply:
separate PR